### ISSUE TITLE
Connection Failed

### ISSUE THREAD

**Issue by techhysahil** [RUB-NDS/PRET#15]
Every time i try any option, i get this.
Connection to 198.16.168.44 failed (Connection refused)
Connection closed.

**Comment by jensvoid** [comment]
I assume you can 198.16.168.44 is your printer and you can successfully ping it.

This error indicates that your printer does not have port 9100 open which is required for PRET to communicate. Port 9100 printing is either deactivated for this device (check the settings in the printer's web interface) or simply not supported.

Which model is it? Can you provide the output of an Nmap scan?

**Comment by techhysahil** [comment]
Yes i try to check for port 9100, It's not Open for the printer. Below is setup for me:

I have a HP Laserjet Professional P1600 & HP Color Laserjet CP1510 series printer connected in my network but 9100 is not open in any. But i am able to print out from my MAC. So how is it connecting & how can i try to get into it.

**Comment by jensvoid** [comment]
Your Mac is probably printing through port 631 (IPP) or port 515 (LPD) (read more on the various printing protocols [here](http://hacking-printers.net/wiki/index.php/Fundamentals#Network_printing_protocols)). If port 9100 is not open -- and if you really cannot enable it in the embedded web server's settings -- you're out of luck and can't use PRET direcly to play with the device because you won't get a direct feedback without port 9100.

You have some other options though:

- Use the `-o` flag to log PRET commands to a file and then print them from USB stick or using `ipptool` or the `lpdprint.py` script included in PRET (you won't get feedback but can you test e.g. DoS attacks or manipulating PS based print jobs).
- Connect through USB A-B cable and use PRET to directly interact with the device (this may be the easiest option)
